app/Jobs/EncodeTrackFile.php

```diff
@@ -181,5 +209,21 @@ public function failed()
         $this->trackFile->status = TrackFile::STATUS_PROCESSING_ERROR;
         $this->trackFile->expires_at = null;
         $this->trackFile->save();
+
+        if ($this->isReplacingTrack) {
+            // If a new version is being uploaded to replace a file, yet the upload fails,
+            // all track files for that version should be deleted as it would other clutter the version
+            if ($this->isForUpload) {
+                $trackFiles = $this->trackFile->track->trackFilesForVersion($this->trackFile->version)->get();
+                foreach ($trackFiles as $trackFile) {
+                    if (File::exists($trackFile->getFile())) {
+                        File::delete($trackFile->getFile());
+                    }
+                    $trackFile->delete();
+                }
+            }
+            $this->trackFile->track->version_upload_status = Track::STATUS_ERROR;
```

Comment:
Will the track continue "working" with the original track version if this happens?